cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -921,31 +921,38 @@ public List<ProgramHistory> getRuns(Collection<ProgramId> programs, ProgramRunSt
   @Override
   public Optional<byte[]> getState(AppStateKey request) throws ApplicationNotFoundException {
     return TransactionRunners.run(transactionRunner, context -> {
-      verifyApplicationExists(request.getNamespaceId(), request.getAppName());
+      ApplicationMeta latest = getLatest(request.getNamespaceId(), request.getAppName());
+      verifyApplicationExists(request.getNamespaceId(), request.getAppName(),
+                              latest == null ? null : latest.getSpec().getAppVersion());
```

Comment:
this is to ensure that the latest version is picked up for the app and not -SNAPSHOT @katukota-google

Reply:
Don't quite understand the logic here, if latest == null, this should already mean the app does not exist, you should move the getLatest logic into `verifyApplicationExists` and refactor the logic there